test/PowerShellEditorServices.Test/Language/SemanticTokenTest.cs

```diff
@@ -0,0 +1,190 @@
+        {
+            string text = "\"$(@($Array).Count) OtherText\"";
+            ScriptFile scriptFile = new ScriptFile(
+                // Use any absolute path. Even if it doesn't exist.
+                DocumentUri.FromFileSystemPath(Path.Combine(Path.GetTempPath(), "TestFile.ps1")),
+                text,
+                Version.Parse("5.0"));
+
+            foreach (Token t in scriptFile.ScriptTokens)
+            {
+                List<SemanticToken> mappedTokens = new List<SemanticToken>(PsesSemanticTokensHandler.ConvertToSemanticTokens(t));
+                switch (t.Text)
+                {
+                    case "$Array":
+                        Assert.Collection(mappedTokens, sToken => Assert.Equal(SemanticTokenType.Variable, mappedTokens[0].Type));
+                        break;
+                    case "Count":
+                        Assert.Collection(mappedTokens, sToken => Assert.Equal(SemanticTokenType.Member, mappedTokens[0].Type));
+                        break;
+                }
+            }
+        }
+
+        [Fact]
+        public async Task RecognizesCurlyQuotedString()
+        {
+            string text = "“^[-'a-z]*”";
+            ScriptFile scriptFile = new ScriptFile(
+                // Use any absolute path. Even if it doesn't exist.
+                DocumentUri.FromFileSystemPath(Path.Combine(Path.GetTempPath(), "TestFile.ps1")),
+                text,
+                Version.Parse("5.0"));
+
+            List<SemanticToken> mappedTokens = new List<SemanticToken>(PsesSemanticTokensHandler.ConvertToSemanticTokens(scriptFile.ScriptTokens[0]));
+            Assert.Collection(mappedTokens, sToken => Assert.Equal(SemanticTokenType.String, mappedTokens[0].Type));
+        }
+
+        [Fact]
+        public async Task RecognizeDoubleQuotedHereString()
+        {
+            string text =  @"
+enum MyEnum{
+    one
+    two
+    three
+}
+";
+            ScriptFile scriptFile = new ScriptFile(
+                // Use any absolute path. Even if it doesn't exist.
+                DocumentUri.FromFileSystemPath(Path.Combine(Path.GetTempPath(), "TestFile.ps1")),
+                text,
+                Version.Parse("5.0"));
+
+            foreach (Token t in scriptFile.ScriptTokens)
+            {
+                List<SemanticToken> mappedTokens = new List<SemanticToken>(PsesSemanticTokensHandler.ConvertToSemanticTokens(t));
+                switch (t.Text)
+                {
+                    case "enum":
+                        Assert.Collection(mappedTokens, sToken => Assert.Equal(SemanticTokenType.Keyword, mappedTokens[0].Type));
+                        break;
+                    case "MyEnum":
+                    case "one":
+                    case "two":
+                    case "three":
+                        Assert.Collection(mappedTokens, sToken => Assert.Equal(SemanticTokenType.Member, mappedTokens[0].Type));
+                        break;
+                }
```

Comment:
Change the switch to using a full `Assert.Collection`

Reply:
the problem with the switch is that:

1. you're not checking the count, so if mappedTokens for some reason has an extra `MyEnum` in there, the test will still pass even though it should fail
2. you're not checking for a scenario where you get a random other token (aka a switch default case) if there's a random token with text "foo" in there, your test will pass when it should fail

Using `Assert.Collection` can completely replace the switch and cover those^ two points as well.

Reply:
Hm I guess it's sort of unclear what's happening in the test but the original script is
enum MyEnum{
    one
    two
    three
}
and the test iterates through all the AST tokens (enum MyEnum { one two three }). The switch statements check the tokens that are known to be converted to SemanticTokenType (everything but the braces).

Reply:
Oh I see! In that case, that makes sense :)